tests/rados/pool_tests.py

```diff
@@ -400,3 +401,44 @@ def run(ceph_cluster, **kw):
             log.info(f"Completed the test of pg_min_num on pool: {entry['pool_name']} ")
         log.info("pg_min_num tests completed")
         return 0
+
+    if config.get("verify_pg_num_limit"):
+        """
+        This test is to verify the pg number limit for a pools.
+        Script cover the following steps-
+           1. Creating a pool with default pg number
+           2. set the Logs
+           3. Set noautoscale flag on
+           4. Set the debug_mgr to 10
+           5. Set the pg number more than the limit that is more than 128
+           6. Check the logs for the proper message
+        """
+        # enable the file logging
+        if not rados_obj.enable_file_logging():
+            log.error("Error while setting config to enable logging into file")
+            return 1
+        log.info("Logging to file configured")
+        pool_name = config.get("verify_pg_num_limit")["pool_name"]
+        rados_obj.create_pool(pool_name)
+
+        # Setting the no-autoscale flag
+        cmd = f'{"ceph osd pool set noautoscale"}'
+        rados_obj.run_ceph_command(cmd=cmd)
+
+        # setting the debug_mgr to 10
+        cmd = f'{"ceph tell mgr config set debug_mgr 10"}'
+        rados_obj.run_ceph_command(cmd=cmd)
+
+        # set the pg number more than 128
+        cmd = f'{"ceph osd pool set {pool_name} pg_num 256"}'
+        rados_obj.run_ceph_command(cmd=cmd)
+        time.sleep(10)
+
+        # checking the logs
+        cmd = f'{"grep -rnw /var/log/ceph/ -e pg_num_actual"}'
+        rados_obj.run_ceph_command(cmd=cmd)
```

Comment:
Can we verify if the grep returned the desired lines as expected?
```
 ./ceph-mgr.depressa010.ceph.redhat.com.nnsjkl.log:39415:2022-06-20T08:40:08.290+0000 7ff90c226700 10 mgr.server operator() pool 5 pg_num 17 - pgp_num 16 gap > max_pg_num_change 1 - must scale pgp_num first
./ceph-mgr.depressa010.ceph.redhat.com.nnsjkl.log:39732:2022-06-20T08:40:20.298+0000 7ff90c226700 10 mgr.server operator() pool 5 pg_num 18 - pgp_num 17 gap > max_pg_num_change 1 - must scale pgp_num first
 ```

Reply:
Ideally, this cmd should be run via `ShellMixin.shell()` since this grep is not a ceph command. 

Checking the logs, that you can see the error:
``` 
Running command (timeout=None): /usr/bin/podman run --rm --ipc=host --net=host --privileged --group-add=disk --init -i -e CONTAINER_IMAGE=registry-proxy.engineering.redhat.com/rh-osbs/rhceph@sha256:527a7fc8e721632940b68c1344c7908513786e3eb6b85f7297fe0988136fb10d -e NODE_NAME=ceph-rados-regression-oiikdk-node1-installer -e CEPH_USE_RANDOM_NONCE=1 -v /var/run/ceph/d2df886c-075d-11ed-8c5d-fa163ecdeb02:/var/run/ceph:z -v /var/log/ceph/d2df886c-075d-11ed-8c5d-fa163ecdeb02:/var/log/ceph:z -v /var/lib/ceph/d2df886c-075d-11ed-8c5d-fa163ecdeb02/crash:/var/lib/ceph/crash:z -v /dev:/dev -v /run/udev:/run/udev -v /sys:/sys -v /run/lvm:/run/lvm -v /run/lock/lvm:/run/lock/lvm -v /var/lib/ceph/d2df886c-075d-11ed-8c5d-fa163ecdeb02/selinux:/sys/fs/selinux:ro -v /:/rootfs -v /etc/ceph/ceph.conf:/etc/ceph/ceph.conf:z -v /etc/ceph/ceph.client.admin.keyring:/etc/ceph/ceph.keyring:z --entrypoint grep registry-proxy.engineering.redhat.com/rh-osbs/rhceph@sha256:527a7fc8e721632940b68c1344c7908513786e3eb6b85f7297fe0988136fb10d -rnw /var/log/ceph/ -e pg_num_actual -f json
grep: json: No such file or directory
 10.0.210.50
2022-07-19 08:39:57,106 - INFO - cephci.pool_tests:110 - Target ratio tests completed 
```

Reply:
Modified the code as per comments. 

Here is the successful log - 
----------------------------------
022-07-22 00:34:20,152 - INFO - cephci.pool_tests:110 - Running command cephadm -v shell -- grep -rnw /var/log/ceph/ -e pg_num_actual on 10.0.209.247 timeout 600
2022-07-22 00:34:22,069 - INFO - cephci.pool_tests:110 - Command completed successfully
2022-07-22 00:34:22,070 - INFO - cephci.pool_tests:110 - Target ratio tests completed
2022-07-22 00:34:22,070 - INFO - cephci.pool_tests:110 - Running command podman --version | awk {'print $3'} on 10.0.209.247 timeout 600
2022-07-22 00:34:22,213 - INFO - cephci.pool_tests:110 - Podman Version 4.0.2
2022-07-22 00:34:22,214 - INFO - cephci.pool_tests:110 - Running command docker --version | awk {'print $3'} on 10.0.209.247 timeout 600
2022-07-22 00:34:22,311 - INFO - cephci.pool_tests:110 - Running command ceph --version | awk '{print $3}' on 10.0.208.90 timeo

Failure log-
-------------
2022-07-22 00:45:43,550 - INFO - cephci.pool_tests:110 - Running command cephadm -v shell -- ceph config set global log_to_file true on 10.0.209.247 timeout 600
2022-07-22 00:45:47,071 - INFO - cephci.pool_tests:110 - Command completed successfully
2022-07-22 00:45:47,074 - INFO - cephci.pool_tests:110 - Running command cephadm -v shell -- ceph config set global mon_cluster_log_to_file true on 10.0.209.247 timeout 600
2022-07-22 00:45:50,643 - INFO - cephci.pool_tests:110 - Command completed successfully
2022-07-22 00:45:50,644 - INFO - cephci.pool_tests:110 - Logging to file configured
2022-07-22 00:45:50,644 - INFO - cephci.pool_tests:110 - Running command cephadm -v shell -- ceph osd pool create pool_num_chk on 10.0.209.247 timeout 600
2022-07-22 00:45:53,442 - INFO - cephci.pool_tests:110 - Command completed successfully
2022-07-22 00:45:53,442 - INFO - cephci.pool_tests:110 - Running command cephadm -v shell -- sudo ceph osd pool application enable pool_num_chk rados on 10.0.209.247 timeout 600
2022-07-22 00:45:57,111 - INFO - cephci.pool_tests:110 - Command completed successfully
2022-07-22 00:45:57,111 - INFO - cephci.pool_tests:110 - Created pool pool_num_chk successfully
2022-07-22 00:45:57,112 - INFO - cephci.pool_tests:110 - Running command cephadm -v shell -- ceph osd pool set noautoscale -f json on 10.0.209.247 timeout 300
2022-07-22 00:46:00,151 - INFO - cephci.pool_tests:110 - Command completed successfully
2022-07-22 00:46:00,152 - INFO - cephci.pool_tests:110 - Running command cephadm -v shell -- ceph tell mgr config set debug_mgr 10 -f json on 10.0.209.247 timeout 300
2022-07-22 00:46:03,049 - INFO - cephci.pool_tests:110 - Command completed successfully
2022-07-22 00:46:03,049 - INFO - cephci.pool_tests:110 - Running command cephadm -v shell -- ceph osd pool set pool_num_chk pg_num 256 -f json on 10.0.209.247 timeout 300
2022-07-22 00:46:07,062 - INFO - cephci.pool_tests:110 - Command completed successfully
2022-07-22 00:46:17,063 - INFO - cephci.pool_tests:110 - Running command cephadm -v shell -- grep -rnw /var/log/ceph/ -e Bharath on 10.0.209.247 timeout 600
2022-07-22 00:46:19,401 - ERROR - cephci.pool_tests:110 - Error 1 during cmd, timeout 600
2022-07-22 00:46:19,402 - ERROR - cephci.pool_tests:110 - Logs are not generated with pg_num_actual
2022-07-22 00:46:19,403 - ERROR - cephci.pool_tests:110 - cephadm -v shell -- grep -rnw /var/log/ceph/ -e Bharath Error:  --------------------------------------------------------------------------------
cephadm ['-v', 'shell', '--', 'grep', '-rnw', '/var/log/ceph/', '-e', 'Bharath']
/usr/bin/podman: 4.0.2
Using default config: /etc/ceph/ceph.conf
/usr/bin/podman: 0426da55c359,20.31MB / 3.911GB
/usr/bin/podman: 21260c2dc474,131.8MB / 3.911GB
/usr/bin/podman: 3587067f146a,78.57MB / 3.911GB
/usr/bin/podman: 520395f20103,23.37MB / 3.911GB
/usr/bin/podman: 9b663a61f638,469.6MB / 3.911GB
/usr/bin/podman: cec8380bebda,35.39MB / 3.911GB
/usr/bin/podman: 0426da55c359,2.81%
/usr/bin/podman: 21260c2dc474,1.83%
/usr/bin/podman: 3587067f146a,4.77%
/usr/bin/podman: 520395f20103,1.42%
/usr/bin/podman: 9b663a61f638,6.62%
/usr/bin/podman: cec8380bebda,5.64%
/usr/bin/podman: 0426da55c359,2.81%
/usr/bin/podman: 21260c2dc474,1.83%
/usr/bin/podman: 3587067f146a,4.77%
/usr/bin/podman: 520395f20103,1.42%
/usr/bin/podman: 9b663a61f638,6.62%
/usr/bin/podman: cec8380bebda,5.64%
Inferring fsid 98c77bac-0975-11ed-975b-fa163ee32b49
Using specified config: /etc/ceph/ceph.conf
/usr/bin/podman: registry-proxy.engineering.redhat.com/rh-osbs/rhceph@sha256:31dd7d8fa99d765657df7e1b660b3441a5dd90115265c8379d762765e86a5651
Using recent ceph image registry-proxy.engineering.redhat.com/rh-osbs/rhceph@sha256:31dd7d8fa99d765657df7e1b660b3441a5dd90115265c8379d762765e86a5651
stat: 167 167
sestatus: SELinux status:                 enabled
sestatus: SELinuxfs mount:                /sys/fs/selinux
sestatus: SELinux root directory:         /etc/selinux
sestatus: Loaded policy name:             targeted
sestatus: Current mode:                   enforcing
sestatus: Mode from config file:          enforcing
sestatus: Policy MLS status:              enabled
sestatus: Policy deny_unknown status:     allowed
sestatus: Memory protection checking:     actual (secure)
sestatus: Max kernel policy version:      33
sestatus: SELinux status:                 enabled
sestatus: SELinuxfs mount:                /sys/fs/selinux
sestatus: SELinux root directory:         /etc/selinux
sestatus: Loaded policy name:             targeted
sestatus: Current mode:                   enforcing
sestatus: Mode from config file:          enforcing
sestatus: Policy MLS status:              enabled
sestatus: Policy deny_unknown status:     allowed
sestatus: Memory protection checking:     actual (secure)
sestatus: Max kernel policy version:      33
Running command (timeout=None): /usr/bin/podman run --rm --ipc=host --net=host --privileged --group-add=disk --init -i -e CONTAINER_IMAGE=registry-proxy.engineering.redhat.com/rh-osbs/rhceph@sha256:31dd7d8fa99d765657df7e1b660b3441a5dd90115265c8379d762765e86a5651 -e NODE_NAME=ceph-rados-regression-9q05xm-node1-installer -e CEPH_USE_RANDOM_NONCE=1 -v /var/run/ceph/98c77bac-0975-11ed-975b-fa163ee32b49:/var/run/ceph:z -v /var/log/ceph/98c77bac-0975-11ed-975b-fa163ee32b49:/var/log/ceph:z -v /var/lib/ceph/98c77bac-0975-11ed-975b-fa163ee32b49/crash:/var/lib/ceph/crash:z -v /dev:/dev -v /run/udev:/run/udev -v /sys:/sys -v /run/lvm:/run/lvm -v /run/lock/lvm:/run/lock/lvm -v /var/lib/ceph/98c77bac-0975-11ed-975b-fa163ee32b49/selinux:/sys/fs/selinux:ro -v /:/rootfs -v /etc/ceph/ceph.conf:/etc/ceph/ceph.conf:z -v /etc/ceph/ceph.client.admin.keyring:/etc/ceph/ceph.keyring:z --entrypoint grep registry-proxy.engineering.redhat.com/rh-osbs/rhceph@sha256:31dd7d8fa99d765657df7e1b660b3441a5dd90115265c8379d762765e86a5651 -rnw /var/log/ceph/ -e Bharath

Logs are attached.